src/main/java/org/ibp/api/brapi/v2/germplasm/BreedingMethodResourceBrapi.java

```diff
@@ -0,0 +1,98 @@
+package org.ibp.api.brapi.v2.germplasm;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.generationcp.middleware.api.breedingmethod.BreedingMethodDTO;
+import org.ibp.api.brapi.v1.common.BrapiPagedResult;
+import org.ibp.api.brapi.v1.common.EntityListResponse;
+import org.ibp.api.brapi.v1.common.Metadata;
+import org.ibp.api.brapi.v1.common.Pagination;
+import org.ibp.api.brapi.v1.common.Result;
+import org.ibp.api.brapi.v1.common.SingleEntityResponse;
+import org.ibp.api.domain.common.PagedResult;
+import org.ibp.api.java.breedingmethod.BreedingMethodService;
+import org.ibp.api.rest.common.PaginatedSearch;
+import org.ibp.api.rest.common.SearchSpec;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+@Api(value = "BrAPI Germplasm Services")
+@Controller
+public class BreedingMethodResourceBrapi {
+
+	@Autowired
+	private BreedingMethodService breedingMethodService;
+
+	@ApiOperation(value = "Get the Breeding Methods", notes = "Get the list of germplasm breeding methods available in a system.")
+	@RequestMapping(value = "/{crop}/brapi/v2/breedingmethods", method = RequestMethod.GET)
+	@ResponseBody
+	public ResponseEntity<EntityListResponse<BreedingMethod>> getAllBreedingMethods(@PathVariable final String crop, @ApiParam(value = BrapiPagedResult.CURRENT_PAGE_DESCRIPTION)
+	@RequestParam(value = "page", required = false) final Integer currentPage,
+		@ApiParam(value = BrapiPagedResult.PAGE_SIZE_DESCRIPTION)
+		@RequestParam(value = "pageSize", required = false) final Integer pageSize) {
+		final PagedResult<BreedingMethod> resultPage = new PaginatedSearch().executeBrapiSearch(currentPage, pageSize,
+			new SearchSpec<BreedingMethod>() {
+
+				@Override
+				public long getCount() {
+					return BreedingMethodResourceBrapi.this.breedingMethodService.getAllBreedingMethods().size();
```

Comment:
Implement a` countBreedingMethods(BreedingMethodSearchRequest methodSearchRequest)` in Middleware and BreedingMethodService in BMSAPI and use that here using an empty search filter